solution/index.js

```diff
@@ -0,0 +1,554 @@
+}
+
+// ===> after edit - save changes <===
+function onTaskBlur(event){   
+
+    let target = event.target;
+    let textAfterEdit = target.textContent;
+
+    displayError(false);
+    editFlag = false;
+   
+    if(textAfterEdit === ""){
+        displayError(true, "Error: Empty task is not valid");    
+        target.textContent = SelectedTaskName;
+        return;
+    }
+
+    if (textAfterEdit != SelectedTaskName) {
+        const tasksObject = getLocalStorageTasks();
+        const arrayOfSection = tasksObject[SelectedTask.dataset.section];
+        
+        let taskBeforeEditIndex = arrayOfSection.indexOf(SelectedTaskName);
+        arrayOfSection[taskBeforeEditIndex] = textAfterEdit; 
+        SelectedTaskName = textAfterEdit;        
+        localStorage.setItem('tasks', JSON.stringify(tasksObject));    
+        reloadTasksPage(searchInput.textContent);
+    }
+}
+
+// ===> On right click <===
+function contextMenuTask(event){
+    event.preventDefault();          
+
+    changeSelectedTask(event.target);
+    
+    const { clientX: mouseX, clientY: mouseY } = event;
+    
+    contextMenu.style.top = `${mouseY}px`;
+    contextMenu.style.left = `${mouseX}px`;
+    
+    contextMenu.classList.add("visible");
+
+    document.addEventListener("click", OnContextMenuClick);
+
+}
+
+function OnContextMenuClick(event){    
+    if (event.target.offsetParent == contextMenu){ 
+        if (event.target.dataset.type === "Delete" ){                        
+            if(confirm("Are you sure you want to delete " + SelectedTaskName + " task?")){
+                deleteTask();
+                reloadTasksPage(searchInput.value);  
+            }                  
+        }               
+    }
+    contextMenu.classList.remove("visible");
+    document.removeEventListener("click", OnContextMenuClick);    
+}
+
+function dragEnter(event){
+    event.target.classList.add('droppable-hovered');    
+}
+
+function dragLeave(event){    
+   event.target.classList.remove('droppable-hovered');   
+}
+
+function dragDrop(event){
+    event.target.classList.remove('droppable-hovered');
+    let target = event.target;
+    const taskManagerDataJSON = getLocalStorageTasks(); 
+    const sectionArray = taskManagerDataJSON[SelectedTask.dataset.section];
+    let index = sectionArray.indexOf(SelectedTaskName);
+    // Remove from previous sction
+    sectionArray.splice(index, 1);
+    localStorage.setItem('tasks', JSON.stringify(taskManagerDataJSON));
+    //Add to the new one in a specific index
+    let indexDrop = parseInt(target.dataset.drop_div);    
+    if(target.dataset.section === SelectedTask.dataset.section && indexDrop !== 0 && index < indexDrop){
+        AddToSection(SelectedTask.textContent, target.dataset.section, indexDrop - 1);
+    }
+    else{
+        AddToSection(SelectedTask.textContent, target.dataset.section, target.dataset.drop_div);
+    }
+    reloadTasksPage(searchInput.textContent); 
```

Comment:
There is no need to reboot all the entries once a change has been made.There are more convenient and efficient ways